### PR TITLE
JAMES-1543 Finish implementing incomplete SIZE extended SMTP behaviour

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

File: protocols/smtp/src/main/java/org/apache/james/protocols/smtp/core/esmtp/MailSizeEsmtpExtension.java [hunk 1]
```diff
@@ -148,16 +149,21 @@ private HookResult doMailSize(SMTPSession session,
...
     public Response onLine(SMTPSession session, ByteBuffer line, LineHandler<SMTPSession> next) {
-        Response response = null;
-    	Boolean failed = (Boolean) session.getAttachment(MESG_FAILED, State.Transaction);
+        Boolean failed = (Boolean) session.getAttachment(MESG_FAILED, State.Transaction);
         // If we already defined we failed and sent a reply we should simply
         // wait for a CRLF.CRLF to be sent by the client.
-        if (failed != null && failed.booleanValue()) {
-            // TODO
+        if (failed != null && failed) {
+            if (line.remaining() == 3 && line.get() == 46) {
```
Comment (quynhn) [inline protocols/smtp/src/main/java/org/apache/james/protocols/smtp/core/esmtp/MailSizeEsmtpExtension.java]: what is the 3 and 46?

File: protocols/smtp/src/main/java/org/apache/james/protocols/smtp/core/esmtp/MailSizeEsmtpExtension.java [hunk 2]
```diff
@@ -148,16 +151,21 @@ private HookResult doMailSize(SMTPSession session,
...
-    	Boolean failed = (Boolean) session.getAttachment(MESG_FAILED, State.Transaction);
+        Boolean failed = (Boolean) session.getAttachment(MESG_FAILED, State.Transaction);
         // If we already defined we failed and sent a reply we should simply
         // wait for a CRLF.CRLF to be sent by the client.
-        if (failed != null && failed.booleanValue()) {
-            // TODO
+        if (failed != null && failed) {
+            if (isDataTerminated(line)) {
+                line.rewind();
+                next.onLine(session, ByteBuffer.wrap(".\r\n".getBytes()));
```
Comment (mbaechler) [inline protocols/smtp/src/main/java/org/apache/james/protocols/smtp/core/esmtp/MailSizeEsmtpExtension.java]: you are missing the charset here

Comment (mbaechler) [inline protocols/smtp/src/main/java/org/apache/james/protocols/smtp/core/esmtp/MailSizeEsmtpExtension.java]: and isn't it "newline" anyway ?

File: server/protocols/protocols-smtp/src/test/java/org/apache/james/smtpserver/SMTPServerTest.java [hunk 3]
```diff
@@ -479,6 +479,48 @@ public void testSimpleMailSendWithEHLO() throws Exception {
...
+        init(smtpConfiguration);
+        int maxSize = 1024;
+        smtpServer.setMaximalMessageSize(maxSize);
+
+        SMTPClient smtpProtocol = new SMTPClient();
+        InetSocketAddress bindedAddress = new ProtocolServerUtils(smtpServer).retrieveBindedAddress();
+        smtpProtocol.connect(bindedAddress.getAddress().getHostAddress(), bindedAddress.getPort());
+
+        // no message there, yet
+        assertThat(queue.getLastMail())
```
Comment (mbaechler) [inline server/protocols/protocols-smtp/src/test/java/org/apache/james/smtpserver/SMTPServerTest.java]: you shouldn't assert at that point or you need another test

Comment (chibenwa) [inline server/protocols/protocols-smtp/src/test/java/org/apache/james/smtpserver/SMTPServerTest.java]: What do you mean? That I need to unit test the wall Netty server?

File: protocols/smtp/src/main/java/org/apache/james/protocols/smtp/core/esmtp/MailSizeEsmtpExtension.java (hunk 2 above)
Comment (chibenwa) [inline protocols/smtp/src/main/java/org/apache/james/protocols/smtp/core/esmtp/MailSizeEsmtpExtension.java]: Will try. I just moved what this legacy code was doing.

File: server/protocols/protocols-smtp/src/test/java/org/apache/james/smtpserver/SMTPServerTest.java (hunk 3 above)
Comment (mbaechler) [inline server/protocols/protocols-smtp/src/test/java/org/apache/james/smtpserver/SMTPServerTest.java]: It's not new that a good test is prepare/run/assert.
You are mixing assertions, so you either need another test for that (no matter what it means) or you doesn't need this assert.
Usually, the assert should be similar to the test title.

File: protocols/smtp/src/main/java/org/apache/james/protocols/smtp/core/esmtp/MailSizeEsmtpExtension.java (hunk 2 above)
Comment (mbaechler) [inline protocols/smtp/src/main/java/org/apache/james/protocols/smtp/core/esmtp/MailSizeEsmtpExtension.java]: Yes, but it's a bug and we spotted it, it doesn't cost much at this point to fix it

File: server/protocols/protocols-smtp/src/test/java/org/apache/james/smtpserver/SMTPServerTest.java (hunk 3 above)
Comment (chibenwa) [inline server/protocols/protocols-smtp/src/test/java/org/apache/james/smtpserver/SMTPServerTest.java]: Ok. Got it.

**Comment by chibenwa** [comment]
```
[Building Java 8 james-1543-2-e15e6ca3-8a85-495e-b61e-f90ee7eca93a] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 33.963 sec <<< FAILURE! - in org.apache.james.jmap.ContainerTest
[Building Java 8 james-1543-2-e15e6ca3-8a85-495e-b61e-f90ee7eca93a] containerShouldBeReachableOnExposedPort(org.apache.james.jmap.ContainerTest)  Time elapsed: 7.454 sec  <<< ERROR!
[Building Java 8 james-1543-2-e15e6ca3-8a85-495e-b61e-f90ee7eca93a] org.apache.http.conn.HttpHostConnectException: Connect to 172.17.0.12:80 [/172.17.0.12] failed: Connection refused
[Building Java 8 james-1543-2-e15e6ca3-8a85-495e-b61e-f90ee7eca93a] 	at org.apache.james.jmap.ContainerTest.containerShouldBeReachableOnExposedPort(ContainerTest.java:45)
[Building Java 8 james-1543-2-e15e6ca3-8a85-495e-b61e-f90ee7eca93a] Caused by: java.net.ConnectException: Connection refused
[Building Java 8 james-1543-2-e15e6ca3-8a85-495e-b61e-f90ee7eca93a] 	at org.apache.james.jmap.ContainerTest.containerShouldBeReachableOnExposedPort(ContainerTest.java:45)
[Building Java 8 james-1543-2-e15e6ca3-8a85-495e-b61e-f90ee7eca93a] 
[Building Java 8 james-1543-2-e15e6ca3-8a85-495e-b61e-f90ee7eca93a] 
[Building Java 8 james-1543-2-e15e6ca3-8a85-495e-b61e-f90ee7eca93a] Results :
[Building Java 8 james-1543-2-e15e6ca3-8a85-495e-b61e-f90ee7eca93a] 
[Building Java 8 james-1543-2-e15e6ca3-8a85-495e-b61e-f90ee7eca93a] Tests in error: 
[Building Java 8 james-1543-2-e15e6ca3-8a85-495e-b61e-f90ee7eca93a]   ContainerTest.containerShouldBeReachableOnExposedPort:45 » HttpHostConnect Con...
```

test this please

**Comment by chibenwa** [comment]
test this please

**Comment by chibenwa** [comment]
test this please

**Comment by chibenwa** [comment]
```
[Building Java 8 james-1543-2-80e063ae-ceb6-46ce-be40-ce899945ef79] [INFO] Apache James MPT Core .............................. FAILURE [01:03 min]
```

test this please

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**